Added new kb article tilelayout-kb-nested-instances (#2503)
telerik/blazor-docs: knowledge-base/tilelayout-nested-instances.md

---

> title: Nested TileLayout Components with Resizing and Reordering

Reviewer: ```suggestion
title: Nesting TileLayout Components with Resizing and Reordering
```

description: Learn how to nest TileLayout components in a Blazor application and enable the resize and reorder features.
type: how-to
page_title: How to Handle Nested Telerik TileLayouts with Resizing and Reordering
slug: tilelayout-kb-nested-instances
tags: tilelayout, blazor, nested, resize, reorder
res_type: kb
ticketid: 1666719
---

## Environment

<table>
<tbody>
<tr>
<td>Product</td>
<td>TileLayout for Blazor
</tr>
</tbody>
</table>

## Description

> I have nested TileLayout components and I want both instances to be resizable and reorderable. The problem is that when I enable these features, the TileLayout does not behave well - for example, the reorder in the child component will also reorder the parent component with the same order.

Reviewer: ```suggestion
I have nested TileLayout components and I want both instances to be resizable and reorderable. The problem is that when I enable these features, the TileLayout does not behave well&mdash;for example, the reorder in the child component will also reorder the parent component with the same order.
```

This KB article answers the following questions:
- How can I enable resizing for nested TileLayout components without affecting the parent component?
- What is the best practice for managing nested TileLayout components in Blazor?
- How to prevent nested TileLayout components from resizing or reordering their parent in Blazor?

## Solution

When using two nested [TileLayout](https://docs.telerik.com/blazor-ui/components/tilelayout/overview) components in a Blazor application, resizing or reordering the child component expectedly affects the parent component. This behavior occurs because the resize and reorder events propagate through both levels of TileLayout components. 

> Nesting TileLayouts is uncommon, but generally possible if the resizing and reordering are enabled only for one level at a time. It is important to clarify that with the users, so they do not expect they can drag tiles across component instances, which is not possible.

Reviewer: ```suggestion
Nesting TileLayouts is uncommon, but generally possible if the resizing and reordering are enabled only for one level at a time. With such an implementation, you need to clarify to your end users that they can't drag tiles across the different component instances.
```

To manage nested TileLyouts with resizing and reordering:

> 1. Bind the `Resizable` and `Reorderable` of both TileLayout instances to different variables, so you can toggle them during runtime.

Reviewer: ```suggestion
1. Bind the `Resizable` and `Reorderable` properties of both TileLayout instances to different variables, so you can toggle them during runtime.
```

> 1. Choose your preferred UI to allow the user enable the resizing and reordering of the specific level of tiles. The example below uses ToggleButtons.

Reviewer: ```suggestion
1. Choose your preferred UI to allow the end user to enable the resizing and reordering of the specific level of tiles. The example below uses ToggleButtons.
```

> 1. Once the user enables the resizing and reordering for one level, programmatically disable the features for the other level.

Reviewer: ```suggestion
1. After the end user enables the resizing and reordering for one level, programmatically disable the features for the other level.
```

> 1. (Optional) Use CSS to manage the overflow of the parent TileLayout. This will prevent the parent TileLayout from expanding if the user resizes the tiles in the child TileLayout.

Reviewer: Recently, you have been switching to lowercase for the optional prefix.

```suggestion
1. (optional) Use CSS to manage the overflow of the parent TileLayout. This prevents the parent TileLayout from expanding if the user resizes the tiles in the child TileLayout.
```

> >caption Enable resize and reorder for only level of nested TileLayouts

Reviewer: ```suggestion
>caption Enable resize and reorder for only one level of nested TileLayouts
```

````CSHTML
<style>
    .parent-tilelayout .k-tilelayout-item {
        overflow: auto;
    }

    .child-tilelayout-disabled .k-tilelayout-item-header {
        pointer-events: none;
    }
</style>

> <TelerikToggleButton @bind-Selected="@ParentOperationsEnabled" Icon="@SvgIcon.Ungroup" OnClick="@EnableParentConfiguration">Enable parent operations</TelerikToggleButton>

Reviewer 2: Actually, we can use only two toggle buttons - one for the parent TileLayout and one for all child tiles. The button for the child tiles can be next to the button for the parent tiles. This will make the example simpler.

Reviewer 3: Synced with @dimodi offline and agreed to keep the sample as is.
